routes/index.js

```diff
@@ -99,6 +99,10 @@ router.post('/login', csrfProtection, loginValidators,
       // Attempt to get the user by their user name!
       const user = await db.User.findOne({ where: { username } });
 
+      if(!user){
+        errors.push('Login failed for the provided user name and password, quest-taker')
+      }
+
```

Comment:
I think you don't need this if statement because you have the else statement that pushes the same error message. Please be sure to remove this and ensure that this works correctly.

Reply:
That if block is needed so that if a user doesn't exist an error is rendered. It is similar to the else statement in the message is the same because it would be bad security to let the hacker know the username exists. The else block error handles if the username exists but the password doesn't match. I think the code could be refactored so that the two different errors(non-existing username and incorrect password) are handled at the same time, if that it what you would like, but deleting lines 102-104 produces a bug.

Reply:
Ohhh I see, thank you for clarifying. Yes in the case, please refactor this so that a non-existent username and an incorrect password are handled at the same time.